### PR TITLE
'userComment' passthrough in Orchestration - BW-761

### PR DESCRIPTION
**Pull request by kpierre13**
\<your comments for this PR go here\>

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [x] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by cjllanwarne** [comment]
@kpierre13 looks a couple of tests were destabilized by this. I'm not 100% sure why your changes did this, but worth looking into:
```
[info] *** 3 TESTS FAILED ***
[error] Failed: Total 747, Failed 3, Errors 0, Passed 744
[error] Failed tests:
[error] 	org.broadinstitute.dsde.firecloud.webservice.WorkspaceApiServiceSpec
[error] 	org.broadinstitute.dsde.firecloud.webservice.SubmissionApiServiceNegativeSpec
```

File: src/main/resources/swagger/api-docs.yaml
Comment (cjllanwarne) [inline src/main/resources/swagger/api-docs.yaml:5921]: Maybe just "Internal Error". Not necessarily from Rawls (even though... probably...)

File: src/main/resources/swagger/api-docs.yaml
Comment (cjllanwarne) [inline src/main/resources/swagger/api-docs.yaml:9064]: Maybe worth matching the `description` of user comment below?
```suggestion
          description: Freeform user defined description, optional.
```

File: src/main/resources/swagger/api-docs.yaml
Comment (cjllanwarne) [inline src/main/resources/swagger/api-docs.yaml:9557]: Is it worth mentioning this 1000 character limit in the docs for the initial submission endpoint too?

File: src/test/scala/org/broadinstitute/dsde/firecloud/webservice/SubmissionApiServiceNegativeSpec.scala
Comment (cjllanwarne) [inline src/test/scala/org/broadinstitute/dsde/firecloud/webservice/SubmissionApiServiceNegativeSpec.scala:57]: Since we're adding a new endpoint (rather than modifying an existing one), we need an OK from security/compliance before merging. I've followed up on our previous request for a compliance review in the ticket here: https://broadworkbench.atlassian.net/browse/BW-761?focusedCommentId=50246

**Review by cjllanwarne — APPROVED** [review]
LGTM with a few small nits. We should wait for a 👍  from appsec/compliance in the jira ticket before merging.

File: src/test/scala/org/broadinstitute/dsde/firecloud/mock/MockWorkspaceServer.scala
Comment (cahrens) [inline src/test/scala/org/broadinstitute/dsde/firecloud/mock/MockWorkspaceServer.scala:68]: I am still trying to understand what sort of unit test coverage makes sense in the orchestration level. @salonishah11 , is this test change (and the one below) sufficient to cover the new end points?

Comment (salonishah11) [reply src/test/scala/org/broadinstitute/dsde/firecloud/mock/MockWorkspaceServer.scala:68]: I think these 2 unit tests cover the addition of userComment in create submission API. We should add unit tests for the new PATCH userComment submission API.

File: src/main/scala/org/broadinstitute/dsde/firecloud/utils/RestJsonClient.scala
Comment (cjllanwarne) [inline src/main/scala/org/broadinstitute/dsde/firecloud/utils/RestJsonClient.scala]: This `Await` is no longer needed and should be removed

Comment (cjllanwarne) [reply src/main/scala/org/broadinstitute/dsde/firecloud/utils/RestJsonClient.scala]: (I suspect it's from me originally, sorry!)

File: src/test/scala/org/broadinstitute/dsde/firecloud/mock/MockWorkspaceServer.scala
Comment (cjllanwarne) [inline src/test/scala/org/broadinstitute/dsde/firecloud/mock/MockWorkspaceServer.scala:91]:
```suggestion
    // Mapping from submission ID to the status code and response which the mock server should return when the PATCH endpoint is called
    val submissionIdPatchResponseMapping = List(
```

File: src/test/scala/org/broadinstitute/dsde/firecloud/webservice/SubmissionApiServiceSpec.scala
Comment (cjllanwarne) [inline src/test/scala/org/broadinstitute/dsde/firecloud/webservice/SubmissionApiServiceSpec.scala]: I don't think this is used any more. We can probably remove it

File: src/test/scala/org/broadinstitute/dsde/firecloud/webservice/SubmissionApiServiceSpec.scala
Comment (cjllanwarne) [inline src/test/scala/org/broadinstitute/dsde/firecloud/webservice/SubmissionApiServiceSpec.scala]:
```suggestion
          (Patch(submissionIdPath, "PATCH request body. The mock server will ignore this content and respond entirely based on submission ID instead")
```

**Review by cjllanwarne — APPROVED** [review]
Re-approved (still pending appsec in jira though, AFAIK?)

**Comment by kpierre13** [comment]
> Re-approved (still pending appsec in jira though, AFAIK?)

I thought appsec approved after the first commit. Should we reach out again with our newest changes?

**Comment by cjllanwarne** [comment]
>I thought appsec approved after the first commit. Should we reach out again with our newest changes?

Oh, I think I just missed that. If you got the thumbs up then great!

File: src/main/resources/swagger/api-docs.yaml
Comment (salonishah11) [inline src/main/resources/swagger/api-docs.yaml]: It would be good to mention the 1000 character limit here as well

File: src/main/scala/org/broadinstitute/dsde/firecloud/utils/RestJsonClient.scala
Comment (salonishah11) [inline src/main/scala/org/broadinstitute/dsde/firecloud/utils/RestJsonClient.scala]: Since this is the only change in the file, it would be better to revert it

Comment (cjllanwarne) [reply src/main/scala/org/broadinstitute/dsde/firecloud/utils/RestJsonClient.scala]: This one was probably my fault too! 😳

**Review by salonishah11 — APPROVED** [review]
LGTM 👍

**Pull request merged**
